streams/src/main/java/org/apache/kafka/streams/kstream/Aggregator.java

```diff
@@ -18,13 +18,22 @@
 package org.apache.kafka.streams.kstream;
 
 /**
- * The Aggregator interface for aggregating values of the given key.
+ * The {@link Aggregator} interface for aggregating values of the given key.
  *
  * @param <K>   key type
  * @param <V>   original value type
  * @param <T>   aggregate value type
  */
 public interface Aggregator<K, V, T> {
 
+    /**
+     * Compute a new aggregate for the message with the given key and value and the current aggregate (ie,
+     * {@link #apply(Object, Object, Object)} "increments" an aggregate with another message).
```

Comment:
From my understanding, `apply` is called for each record to update the current (intermediate) agg-value. Something like this (assume the aggregation is a sum for input `3,2,8` and initial agg-value `0`)

`apply(x, 3, 0) => 3`
`apply(x, 2, 3) => 5`
`apply(x, 8, 5) => 11`

Thus, the sum in incrementally build over all calls to `apply()`. In my understanding, increment does not mean `+1`, but has a more general meaning "next step of the aggregation" -- I also put it into quotes as a try to make this clear. But my understanding might not be common sense, and thus, "increments" might be miss leading.

I thought it might be good to explain the user, that the computed/returned agg-result will be handed in as a parameter in the next call -- thus I put this additional explanation. If you think it is more miss leasing than helpful, we should just remove it. Or maybe you have a better idea to phrase this? (Or is my understanding of `apply()` wrong?)